pkg/controllers/proxyconfig/proxyconfig_controller.go

```diff
@@ -52,7 +52,7 @@ func NewProxyConfigChecker(
 		WithInformers(
 			routeInformer.Informer(),
 		).
-		ResyncEvery(5 * time.Minute).
+		ResyncEvery(60 * time.Minute).
 		WithSyncDegradedOnError(operatorClient)
 
 	for ns := range caConfigMaps {
```

Comment:
This `for` could be quite problematic as it registers event handlers for any CM in a namespace, and so even with a more lean resync, you might fire the sync loop much more often.

Use `c.WithFilteredEventsInformers()` and base the filter off of the CM names that you store in `caConfigMaps` already

Reply:
This makes sense, thanks for the suggestion. I'm not very familiar with informers and event handlers yet -- is this what you meant? :point_right: https://github.com/openshift/cluster-authentication-operator/pull/608/files#diff-067c8061203aef85d6b9fcc6819fda8a9cc8a490a46f38973e0d7e40e0b1c5c7R60

On a side note, I noticed that there's a [TODO](https://github.com/liouk/cluster-authentication-operator/blob/proxyconfig-check-freq/pkg/controllers/common/config.go#L34) on `common.NamesFilter` that it should be in library-go; and in fact it exists in [`library-go/pkg/controller/factory/eventfilters.go`](https://github.com/openshift/library-go/blob/master/pkg/controller/factory/eventfilters.go#L17) -- should I use that one instead?

If yes, there are a couple of other usages of `common.NamesFilter` that we could replace with `factory.NamesFilter` ([here](https://github.com/liouk/cluster-authentication-operator/blob/proxyconfig-check-freq/pkg/controllers/routercerts/controller.go#L82) and [here](https://github.com/liouk/cluster-authentication-operator/blob/proxyconfig-check-freq/pkg/controllers/oauthclientscontroller/oauthclientscontroller.go#L62-L69)), what do you think?

Reply:
> This makes sense, thanks for the suggestion. I'm not very familiar with informers and event handlers yet -- is this what you meant? point_right https://github.com/openshift/cluster-authentication-operator/pull/608/files#diff-067c8061203aef85d6b9fcc6819fda8a9cc8a490a46f38973e0d7e40e0b1c5c7R60

Yes, that looks like it could be used here, too.

> On a side note, I noticed that there's a [TODO](https://github.com/liouk/cluster-authentication-operator/blob/proxyconfig-check-freq/pkg/controllers/common/config.go#L34) on common.NamesFilter that it should be in library-go; and in fact it exists in [library-go/pkg/controller/factory/eventfilters.go](https://github.com/openshift/library-go/blob/master/pkg/controller/factory/eventfilters.go#L17) -- should I use that one instead?

yes, use that instead, feel free to replace all the other use of the names filter in this repo with it

Reply:
done in a separate commit for the other references